integrations/weaviate/src/haystack_integrations/document_stores/weaviate/document_store.py

```diff
@@ -172,11 +172,15 @@ def client(self):
         if self._client:
             return self._client
 
-        if self._url and self._url.startswith("http") and self._url.endswith(".weaviate.network"):
+        if (
+            self._url
+            and self._url.startswith("http")
+            and (self._url.endswith(".weaviate.network") or self._url.endswith(".weaviate.cloud"))
+        ):
             # We use this utility function instead of using WeaviateClient directly like in other cases
             # otherwise we'd have to parse the URL to get some information about the connection.
             # This utility function does all that for us.
-            self._client = weaviate.connect_to_wcs(
+            self._client = weaviate.connect_to_weaviate_cloud(
```

Comment:
I had a quick look and confirm that weaviate describes `connect_to_weaviate_cloud` as the replacement of `connect_to_wcs` https://weaviate.io/developers/weaviate/client-libraries/python#connection-helper-functions
It was part of a renaming in version 4.6.2 https://weaviate-python-client.readthedocs.io/en/stable/changelog.html#version-4-6-2
👍